File: CodingSamples/Java/Language/VarArgTest/Statistics.java
class DoubleRef {
	public double value;
}

class Statistics {

	/*
	public static double average(double first, double second) {
		return (first + second) / 2;
	}

	public static double average(double first, double second, double third) {
		return (first + second + third) / 3;
	}
	*/

	public static double average(double first, double second, double... other)
	{
		double total = first + second;
		for(double item : other){
			total += item;
		}
		return total / (2 + other.length);
	}

	public static double averageWithDeviation(double first, double second, DoubleRef delta) {
		delta.value = first > second ? (first - second) / 2 : (second - first) / 2;
		return average(first, second);
	}
}

File: CodingSamples/DotNet/Language/ParamArrayTest/DemoApp/Statistics.cs
class Statistics
{
    public static double Average(double first, double second)
    {
        return (first + second) / 2;
    }

    public static double Average(double first, double second, double third)
    {
        return (first + second + third) / 3;
    }

    //an array type (last) parameter declared with 'params' keyword
    //can accept multiple arguments through that array
    public static double Average(double first, double second, params double[] other)
    {
        double total = first + second;
        foreach(double value in other)
        {
            total += value;
        }
        return total / (2 + other.Length);
    }

    //a parameter declared with 'out' keyword can accept
    //a reference (like ref) to an uninitialized argument
    //but this parameter must be initialized before control
    //leaves the method
    public static double AverageWithDeviation(double first, double second, out double delta)
    {
        delta = first > second ? (first - second) / 2 : (second - first) / 2;
        return Average(first, second);
    }
    
}